src/command/dev/protocol/mod.rs

```diff
@@ -1,9 +1,27 @@
+pub use follower::*;
+pub use leader::*;
+pub(crate) use socket::*;
+pub use types::*;
+
 mod follower;
 mod leader;
 mod socket;
 mod types;
 
-pub use follower::*;
-pub use leader::*;
-pub(crate) use socket::*;
-pub use types::*;
+macro_rules! create_socket_name {
+    ($raw_socket_name:expr) => {
+        if GenericFilePath::is_supported() {
+            $raw_socket_name
+                .clone()
+                .to_fs_name::<GenericFilePath>()
+                .unwrap()
```

Comment:
do we want unwraps here or something else? Worried about panics (especially without an error message; we should switch to expect if we can't move to something more robust), but also about panics coming from macros making it super, super hard to track down what went wrong